### ISSUE TITLE
xunit.runner.utility NuGet package documentation missing dependency.

### ISSUE THREAD

**Issue by jameswiseman76** [xunit/xunit#2140]
I was facing the following issue getting **xunit.runner.utility** to execute:

_System.InvalidOperationException: Could not find/load any of the following assemblies: xunit.execution.dotnet.dll_

(I can include code context if required, but am omitting just now as I'm not sure it's relevant)

Upon instructions from the nuget package documenation (https://xunit.net/docs/nuget-packages), I included **xunit.abstractions**, but the issue persisted. 

When I added **xunit.extensibility.execution**, the issue resolved itself. I think this ought to be called out in the docs. Also, can you set this a dependency on the package itself, so installing **xunit.runner.utility** will automatically pull these in?

**Comment by bradwilson** [comment]
This isn't a requirement for a runner; it's a requirement for unit tests. I'm not sure why adding a reference to `xunit.execution.dotnet.dll` made your runner suddenly work, but the behavior you're seeing may be expected. Please provide a repro project so I can verify.

**Comment by jameswiseman76** [comment]
Thank for the quick response.

Oh god, I'm going to have to share my code! I'm just mucking around just now, so please don't judge! ;-)

This the repo:
https://github.com/jameswiseman76/UnitTestRunnerApiAdaptor

The following commit (and latest at the time of writing) demonstrates this.
https://github.com/jameswiseman76/UnitTestRunnerApiAdaptor/commit/2778e3581e07114e7eecd231d3addf94d9920824

The console app **TestApiRunner** class **Program** runs it.

**Comment by bradwilson** [comment]
A couple suggestions as I'm working through this.

1. You should bring in `Microsoft.VisualStudio.TestPlatform.TestFramework` via the NuGet package `MSTest.TestFramework` and not from the shared folder.
2. You should bring in the Visual Studio assemblies using `$(VSINSTALLDIR)` rather than relative paths, which will be broken by different versions & editions of Visual Studio (you're using Enterprise, I'm using Community, so your code wouldn't build for me).

I would change this:

```xml
  <ItemGroup>
    <Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework">
      <HintPath>..\..\..\..\..\..\..\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\TestPlatform\Microsoft.VisualStudio.QualityTools.UnitTestFramework.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.VisualStudio.TestPlatform.Extensions.VSTestIntegration">
      <HintPath>..\..\..\..\..\..\..\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\Extensions\TestPlatform\Extensions\Microsoft.VisualStudio.TestPlatform.Extensions.VSTestIntegration.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.VisualStudio.TestPlatform.ObjectModel">
      <HintPath>..\..\..\..\..\..\..\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\IDE\CommonExtensions\Microsoft\TestWindow\Microsoft.VisualStudio.TestPlatform.ObjectModel.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.VisualStudio.TestPlatform.TestFramework">
      <HintPath>..\..\..\..\..\..\..\Program Files (x86)\Microsoft Visual Studio\Shared\Packages\MSTest.TestFramework.1.2.0\lib\net45\Microsoft.VisualStudio.TestPlatform.TestFramework.dll</HintPath>
    </Reference>
  </ItemGroup>
```

to this:

```xml
  <ItemGroup>
    <PackageReference Include="MSTest.TestFramework" Version="1.2.0" />
  </ItemGroup>

  <ItemGroup>
    <Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework">
      <HintPath>$(VSINSTALLDIR)Common7\IDE\Extensions\TestPlatform\Microsoft.VisualStudio.QualityTools.UnitTestFramework.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.VisualStudio.TestPlatform.Extensions.VSTestIntegration">
      <HintPath>$(VSINSTALLDIR)Common7\IDE\Extensions\TestPlatform\Extensions\Microsoft.VisualStudio.TestPlatform.Extensions.VSTestIntegration.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.VisualStudio.TestPlatform.ObjectModel">
      <HintPath>$(VSINSTALLDIR)Common7\IDE\CommonExtensions\Microsoft\TestWindow\Microsoft.VisualStudio.TestPlatform.ObjectModel.dll</HintPath>
    </Reference>
  </ItemGroup>
```

(This is in TestApiRunner and UnitTestRunnerApiAdapter)

**Comment by bradwilson** [comment]
Looking at what you're trying to do, you're not likely to be successful long-term.

You're trying to dynamically load and run .NET Core test assemblies. We used to do that, and gave up, because assembly resolution is too complex to get solved correctly. It is, in fact, one of the primary drivers of the new v3 architecture that makes test assemblies stand-alone applications rather than class libraries, simply because we absolutely have to let the compiler and runtime do the work to assembly resolution for us.

I was (provisionally) able to make it work, by copying a lot of code from xUnit.net itself into your project. As mentioned above, this code is fragile and doesn't always work. If it works for you, then more power to you; if and when it fails, I won't be able to support you, because this is long-abandoned code, abandoned precisely because it could never be made to work fully.

Here are the changes I made: https://github.com/bradwilson/UnitTestRunnerApiAdaptor/commit/ebeafe4a02686306bb1e29fb6f6383a19c3425fd

Note that the source code under `UnitTestRunnerApiAdaptor/XUnit/FromXunit` is all copyright .NET Foundation (some from xUnit.net, and some from .NET Core). License requirements are listed here: https://github.com/xunit/xunit/blob/v2/license.txt
